rosbag2_storage/include/rosbag2_storage/storage_options.hpp

```diff
@@ -46,6 +46,10 @@ struct StorageOptions
   // Storage specific configuration file.
   // Defaults to empty string.
   std::string storage_config_uri = "";
+
+  // Enable snapshot mode.
+  // Defaults to disabled.
+  bool snapshot_mode = false;
```

Comment:
This should be a `RecordOptions` setting - it is not a property of the Storage , which knows nothing about this (`Play` also uses `StorageOptions`).

Reply:
Decided to leave this in `StorageOptions` for now since `RecordOptions` is not accessible within `SequentialWriter`. May move this to `RecordOptions` in a future refactor